da/avail/avail.go

```diff
@@ -0,0 +1,183 @@
+func (c *DataAvailabilityLayerClient) Start() error {
+
+	c.logger.Info("starting avail Data Availability Layer Client", "baseURL", c.config.ApiURL)
+
+	return nil
+}
+
+// Stop stops DataAvailabilityLayerClient.
+func (c *DataAvailabilityLayerClient) Stop() error {
+
+	c.logger.Info("stopping Avail Data Availability Layer Client")
+
+	return nil
+}
+
+// SubmitBlock submits a block to DA layer.
+func (c *DataAvailabilityLayerClient) SubmitBlocks(ctx context.Context, blocks []*types.Block) da.ResultSubmitBlocks {
+
+	for _, block := range blocks {
+		data, err := block.MarshalBinary()
+		if err != nil {
+			return da.ResultSubmitBlocks{
+				BaseResult: da.BaseResult{
+					Code:    da.StatusError,
+					Message: err.Error(),
+				},
+			}
+		}
+		err = datasubmit.SubmitData(c.config.ApiURL, c.config.Seed, c.config.AppID, data)
+
+		if err != nil {
+			return da.ResultSubmitBlocks{
+				BaseResult: da.BaseResult{
+					Code:    da.StatusError,
+					Message: err.Error(),
+				},
+			}
+		}
+	}
+
+	return da.ResultSubmitBlocks{
+		BaseResult: da.BaseResult{
+			Code:     da.StatusSuccess,
+			Message:  "data submitted successfully",
+			DAHeight: 1,
+		},
+	}
+
+}
+
+// RetrieveBlocks gets the block from DA layer.
+
+func (c *DataAvailabilityLayerClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64) da.ResultRetrieveBlocks {
+
+	blocks := []*types.Block{}
+
+	blockNumber := dataLayerHeight
+
+	appDataURL := fmt.Sprintf(c.config.BaseURL+"/appdata/%d?decode=true", blockNumber)
+
+	response, err := http.Get(appDataURL)
+	if err != nil {
+		return da.ResultRetrieveBlocks{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+	defer func() {
+		_ = response.Body.Close()
+	}()
+
+	responseData, err := io.ReadAll(response.Body)
+	if err != nil {
+		return da.ResultRetrieveBlocks{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	var appDataObject AppData
+	err = json.Unmarshal(responseData, &appDataObject)
+	if err != nil {
+		return da.ResultRetrieveBlocks{
+			BaseResult: da.BaseResult{
+				Code:    da.StatusError,
+				Message: err.Error(),
+			},
+		}
+	}
+
+	txnsByteArray := []byte{}
+	for _, extrinsic := range appDataObject.Extrinsics {
+		txnsByteArray = append(txnsByteArray, []byte(extrinsic)...)
+	}
+
+	block := &types.Block{
+		SignedHeader: types.SignedHeader{
+			Header: types.Header{
+				BaseHeader: types.BaseHeader{
+					Height: blockNumber,
+				},
+				AggregatorsHash: make([]byte, 32),
+			}},
+		Data: types.Data{
+			Txs: types.Txs{txnsByteArray},
+			IntermediateStateRoots: types.IntermediateStateRoots{
+				RawRootsList: nil,
+			},
+		},
+	}
+
+	blocks = append(blocks, block)
+
+	return da.ResultRetrieveBlocks{
+		BaseResult: da.BaseResult{
+			Code:     da.StatusSuccess,
+			DAHeight: uint64(appDataObject.Block),
+			Message:  "block data: " + strings.Join(appDataObject.Extrinsics, " "),
```

Comment:
Is concatenating extrinsics into a message needed, or is it just a convenience for debugging?

Reply:
No @aterentic-ethernal ,It's not needed.I just tried to print the Extrinsics in message.I'll remove it